src/editor/editorinteractive.cc

```diff
@@ -736,117 +755,75 @@ void EditorInteractive::toggle_grid() {
 
 bool EditorInteractive::handle_key(bool const down, SDL_Keysym const code) {
 	if (down) {
-		switch (code.sym) {
```

Comment:
Uhm we trade a switch against a lot of ifs.
Could we not use something like switch (shortcut_from_code(...)) { ...

Reply:
The thing is that one code may be mapped to several shortcuts, e.g. you could define that 'T' opens the tool menu in the editor and the general stats menu in the game. There's a scope-checking system in place so shortcuts in the same scope won't collide, but still this approach wouldn't work here. Shortcuts can have multiple scopes, and in a given place in the code multiple scopes may apply (e.g, here we use the Global, Common and Editor scopes). 
So we have to check for each shortcut whether the pressed key matches it, and since the compiler requires that `case` labels can be evaluated at compile-time a `switch` is not applicable here sadly.